azure/scope/managedmachinepool.go

```diff
@@ -240,9 +240,18 @@ func buildAgentPoolSpec(managedControlPlane *infrav1.AzureManagedControlPlane,
 		}
 	}
 
+	if managedControlPlane.Spec.EnablePreviewFeatures != nil {
+		agentPoolSpec.Preview = *managedControlPlane.Spec.EnablePreviewFeatures
+	}
```

Comment:
This should be equivalent without needing the condition:
```suggestion
	agentPoolSpec.Preview = ptr.Deref(managedControlPlane.Spec.EnablePreviewFeatures, false)
```

In which case we might as well define that above where we first construct `agentPoolSpec`.